l3kernel/l3intarray.dtx

```diff
@@ -173,7 +173,454 @@
 %<@@=intarray>
 %    \end{macrocode}
 %
-% \subsection{Allocating arrays}
+% There are two implementations for this module: One \cs{fontdimen} based one
+% for more traditional \TeX\ engines and a Lua based one for engines with Lua support.
+%
+% Both versions do not allow negative array sizes.
+%    \begin{macrocode}
+%<*tex>
+\msg_new:nnn { kernel } { negative-array-size }
+  { Size~of~array~may~not~be~negative:~#1 }
+%    \end{macrocode}
+%
+% \subsection{Lua implementation}
+% First, let's look at the Lua variant:
+%
+% We select the Lua version if the Lua helpers were defined. This can be detected by
+% the presence of \cs{@@_gset_count:Nw}.
+%
+%    \begin{macrocode}
+\cs_if_exist:NTF \@@_gset_count:Nw
+{
+%</tex>
+%    \end{macrocode}
+%
+% \subsubsection{Allocating arrays}
+%
+% \begin{variable}{\l_@@_loop_int}
+%   A loop index.
+%    \begin{macrocode}
+%<*tex>
+\int_new:N \l_@@_loop_int
+%    \end{macrocode}
+% \end{variable}
+%
+% \begin{variable}{\g_@@_table_int, \l_@@_bad_index_int}
+%   Used to differentiate intarrays in Lua and to record an invalid index.
+%    \begin{macrocode}
+\int_new:N \g_@@_table_int
+\int_new:N \l_@@_bad_index_int
+%    \end{macrocode}
+% \end{variable}
+%
+% \begin{macro}{\s_@@}
```

Comment:
Given that `\s_@@` is not actually a scan mark (which are equal to `\relax`), it should probably be called something else like the `\@@_chk:w` used in `l3fp`.

Reply:
Maybe just `\__intarray:w`?

Reply:
Ok, if you prefer.